tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests2.cs

```diff
@@ -61,64 +61,163 @@ public class OboTests2
+                .AcquireTokenOnBehalfOf(scopes2, userAssertion)
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+            
+            Assert.AreEqual(TokenSource.IdentityProvider, authenticationResult.AuthenticationResultMetadata.TokenSource);
             Assert.IsNotNull(authenticationResult.AccessToken);
+            Assert.IsTrue(!userCacheRecorder.LastAfterAccessNotificationArgs.IsApplicationCache);
+            Assert.IsTrue(userCacheRecorder.LastAfterAccessNotificationArgs.HasTokens);
+            Assert.AreEqual(atHash, userCacheRecorder.LastAfterAccessNotificationArgs.SuggestedCacheKey);
             Assert.AreEqual(TokenSource.IdentityProvider, authenticationResult.AuthenticationResultMetadata.TokenSource);
+            Assert.IsNotNull(
+                userCacheRecorder.LastAfterAccessNotificationArgs.SuggestedCacheExpiry,
+                "When writing the OBO token response, MSAL should ignore the RT and propose expiry");
 
-            authenticationResult = await _confidentialApp.AcquireTokenOnBehalfOf(scopes2, userAssertion)
-                                                         .ExecuteAsync().ConfigureAwait(false);
-
-            Assert.IsNotNull(authenticationResult);
+            authenticationResult = await middletierServiceApp
+                .AcquireTokenOnBehalfOf(scopes2, userAssertion)
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+           
             Assert.IsNotNull(authenticationResult.AccessToken);
             Assert.IsTrue(!userCacheRecorder.LastAfterAccessNotificationArgs.IsApplicationCache);
             Assert.IsTrue(userCacheRecorder.LastAfterAccessNotificationArgs.HasTokens);
             Assert.AreEqual(atHash, userCacheRecorder.LastAfterAccessNotificationArgs.SuggestedCacheKey);
             Assert.AreEqual(TokenSource.Cache, authenticationResult.AuthenticationResultMetadata.TokenSource);
+            Assert.IsNull(
+                userCacheRecorder.LastAfterAccessNotificationArgs.SuggestedCacheExpiry,
+                "The cache expiry is not set because the node did not change");
         }
 
-        [TestInitialize]
-        public void TestInitialize()
+
+        [TestMethod]
+        public async Task ServicePrincipal_OBO_LongRunningProcess_PPE_Async()
         {
-            TestCommon.ResetInternalStaticCaches();
+            //An explanation of the OBO for service principal scenario can be found here 
 
-            if (_keyVault == null)
-            {
-                _keyVault = new KeyVaultSecretsProvider();
-            }
+            var settings = ConfidentialAppSettings.GetSettings(Cloud.Public);
+            var cert = settings.GetCertificate();
+
+            IReadOnlyList<string> scopes = new List<string>() { OBOServicePpeClientID + "/.default" };
+            IReadOnlyList<string> scopes2 = new List<string>() { OBOServiceDownStreamApiPpeClientID + "/.default" };
+
+            
```

Comment:
nit: extra space
Also, there was a PR where I enabled a rule that would highlight extra spaces. Do you recall if that rule was ever merged in?

Reply:
Yes, it looks like it

![Uploading image.png…]()